internal/pkg/utils/testhelper/runner/runner.go

```diff
@@ -25,33 +25,47 @@ import (
 const testTimeout = 5 * time.Minute
 
 type Runner struct {
-	t          *testing.T
-	testsDir   string
-	workingDir string
+	t        *testing.T
+	testsDir string
 }
 
 func NewRunner(t *testing.T) *Runner {
 	t.Helper()
 
 	_, callerFile, _, _ := runtime.Caller(1) //nolint:dogsled
-	callerDir := filepath.Dir(callerFile)    // nolint:forbidigo
+	testsDir := filepath.Dir(callerFile)     // nolint:forbidigo
 
-	workingDir := filesystem.Join(callerDir, ".out")
-	assert.NoError(t, os.RemoveAll(workingDir))
-	assert.NoError(t, os.MkdirAll(workingDir, 0o755))
+	// Delete debug files from the previous run
+	if testhelper.CreateOutDir() {
+		require.NoError(t, os.RemoveAll(filesystem.Join(testsDir, ".out")))
+	}
 
-	return &Runner{t: t, testsDir: callerDir, workingDir: workingDir}
+	return &Runner{t: t, testsDir: testsDir}
 }
 
 func (r *Runner) newTest(t *testing.T, testDirName string) (*Test, context.CancelFunc) {
 	t.Helper()
 
 	testDir := filepath.Join(r.testsDir, testDirName)
-	workingDir := filepath.Join(r.workingDir, testDirName)
 
-	assert.NoError(t, os.RemoveAll(workingDir))
-	assert.NoError(t, os.MkdirAll(workingDir, 0o755))
-	assert.NoError(t, os.Chdir(workingDir))
+	// Create temporary working dir
+	workingDir := t.TempDir()
+	require.NoError(t, os.Chdir(workingDir))
+
+	// Chdir after the test, without it, the deletion of the temp dir is not possible on Windows
+	t.Cleanup(func() {
+		require.NoError(t, os.Chdir(testDir))
+	})
+
+	// Keep working dir for debugging
+	t.Cleanup(func() {
+		if testhelper.CreateOutDir() {
+			outDir := filesystem.Join(r.testsDir, ".out", testDirName)
+			require.NoError(t, os.RemoveAll(outDir))
+			require.NoError(t, os.MkdirAll(outDir, 0o755))
+			require.NoError(t, aferofs.CopyFs2Fs(nil, workingDir, nil, outDir))
```

Comment:
I would spend time on it in different PR where we will replace it everywhere in project. :slightly_smiling_face:

Reply:
Of course you can change it then.

Well, it's not a complete replacement. `aferofs` is a library for virtual file systems, we use it to lock operations only to the working directory (something like a chroot), and we also use in-memory fs. 

But in this place it can be replaced.